docking.js

```diff
@@ -655,23 +654,26 @@ var DockedDash = GObject.registerClass({
     }
 
     _hoverChanged() {
-        if (!this._ignoreHover) {
-            // Skip if dock is not in autohide mode for instance because it is shown
-            // by intellihide.
-            if (this._autohideIsEnabled) {
-                if (this._box.hover)
-                    this._show();
-                else
-                    this._hide();
-            }
+        // Skip if dock is not in autohide mode for instance because it is shown by intellihide or
+        // if the dock's `_ignoreHover` value is set to true.
+        if (this._ignoreHover || !this._autohideIsEnabled) {
+            return;
         }
+
+        if (this._triggerTimeoutId)
+            this._isPointerInZone() || this._box.hover ? this._show() : this._hide();
+        else
+            this._box.hover ? this._show() : this._hide();
```

Comment:
Keeping the `_box.hover` check in an else statement guarantee other docks that do not use `_onPressureSensed` will keep the same behaviour. Because of this, I think setups using intellihide/autohide without the push-to-show option (most likely) will still have the bouncing glitch issue.

Again, I'll leave this part to be reviewed by maintainers with better understanding of the many dash-to-dock's options.